### DISCUSSION TITLE
How to get mean value within data of a facet?

### DISCUSSION THREAD

**Discussion by knutwannheden** [observablehq/plot#1104] [Q&A] [answered]
I am building a faceted strip plot chart in which I would like to display a red rule for the mean value within the facet. I found `Plot.selectMinY()` and `Plot.selectMaxY()`, but I don't understand how I can display the mean value. Tips would be most appreciated.

In the example below I've used `Plot.selectMinY()`:

![image](https://user-images.githubusercontent.com/408765/200682702-e4650a21-4775-42b2-be22-84cb7e743e51.png)

The code (published at https://observablehq.com/d/4a0159d26906f08d) looks like this:
```js
Plot.plot({
  grid: true,
  inset: 10,
  height: 400,
  width: 800,
  facet: {
    data: ib_tr,
    x: "fraktion"
  },
  y: {
    nice: true
  },
  fx: {
    domain: ['G', 'S', 'GL', 'M-E', 'RL', 'V'], 
  },
  marks: [
    Plot.ruleY([0]),
    Plot.ruleY(ib_tr, Plot.selectMinY({y: "rating", stroke: "red"})),
    Plot.dot(ib_tr, Plot.dodgeX({
      y: "rating",
      r: 4,
      title: d =>`${d.parlamentarier}\nMandate: ${d.anzahl}`,
      href: d => `https://lobbywatch.ch/de/daten/parlamentarier/${d.parlamentarier_id}`,
      target: '_blank',
      fill: "fraktion",
      anchor: "middle",
      id: d => d.parlamentarier_id
    }))
  ],
  color: {
    legend: false,
    type: "categorical",
    domain: ['G', 'S', 'GL', 'M-E', 'RL', 'V'], 
    range: ["#85B50C", "#E4022D", "#A0A000", "#FF9100", "#3A8BC1", "#0A7228"]
  },
})
```

**Comment by knutwannheden** [comment]
I think I found a solution or at least a workaround based on the `README.md` in the GitHub repo:
```js
    Plot.ruleY(ib_tr, Plot.select({
      y: (I, V) => {
        const median = d3.mean(I, i => V[i]);
        const i = d3.least(I, i => Math.abs(V[i] - median));
        return [i];
      }
    }, {y: "rating", stroke: "red"})),
```

But AFAICT this will give me the data point that is _nearest_ the mean value, which wasn't exactly what I wanted, but I will use it as a workaround.

**Comment by mbostock** [comment, marked as answer]
You can use the groupZ transform:

```js
Plot.ruleY(ib_tr, Plot.groupZ({y: "median"}, {y: "rating", stroke: "red"}))
```

**Reply by knutwannheden** [reply]
Thanks a lot. This looks exactly like what I wanted!

I still need to wrap my head around this z-channel...

**Reply by knutwannheden** [reply]
The [facets documentation](https://observablehq.com/@observablehq/plot-facets?collection=@observablehq/plot) doesn't appear to mention the `z` channel. Where would I have found the hint for this?

**Reply by mbostock** [reply]
Here there’s no z-channel, so Plot.groupZ means “group everything”. It’s a bit confusing… 🤷 Normally the _z_-channel is used e.g. with a multi-series line chart where you want the lines to the same color.
